### DISCUSSION TITLE
Building examples for CubeMX/Eclipse via CMake

### DISCUSSION THREAD

**Discussion by rsgrimes** [hathach/tinyusb#1579] [Q&A]
In the past, I set up a CMake-based build system for a client who used TI's Eclipse-based Code Composer tools. We wanted to build the executable from the command line (faster, easier to automate), but when it came time to debug, we couldn't really use Code Composer with our project, as Eclipse CDT seems to insist on managing the project. I was able to deal with this by using the -G"Eclipse CDT4 - Unix Makefiles" switch when calling CMake to set up an Eclipse project in the build directory. This worked great!

It would seem I could do the same thing with CubeMX, which is also Eclipse-based, and TinyUSB seems to use CMake. However, I haven't been able to figure out where the TinyUSB build system invokes CMake, so I can't add that switch; the second issue, of course, is how I would do that. Any clues as to where CMake is called, and how could I add extra switches to its invocation? I also will want to send the CROSS_COMPILE switch to make in order to call the correct compiler.

Note: My goal here is to demonstrate TinyUSB on a new chip (STM32U5) that isn't yet supported by TinyUSB. Assuming this ends up working well, we'll be adopting TinyUSB for our project, which will not have to deal with all the clever build magic that TinyUSB provides to build the examples. I'm hoping to submit the STM32U5 mcu and bsp ports for consideration by the TinyUSB project.

**Comment by hathach** [comment]
TinyUSB mainly use makefile, CMAKE is only used for esp32sx and rp2040 since those sdk required cmake to work with. There is another user that detail his step to get cubemx working. You could either find it in tinyusb discussion and/or st community

**Comment by rsgrimes** [comment]
Ah, well that makes sense. When I tried to use CMake, it seemed to only support those MCUs. 

Adding this in case someone else has the need: I realized that the example build system, while quite nice, isn't really what we will want going forward, when we will be incorporating TinyUSB into our projects (and therefore, our build system), so I am taking the route of using the -G"Eclipse CDT4 - Unix Makefiles" CMake magic to create the Eclipse project from our CMakeLists.txt file - the latter of which is straightforward. 

Thanks for the help, Ha!

**Comment by hathach** [comment]
maybe this will help you https://github.com/hathach/tinyusb/discussions/633
